ui/src/js/07-copy-code.js

```diff
@@ -8,7 +11,16 @@
   }
 
   function copyClick(e) {
-    const code = e.target.parentElement.innerText;
+    let code = e.target.parentElement.innerText;
+    const codeParentElement = e.target.parentElement;
+    const codeElement = codeParentElement.querySelector(codeElementSelector);
+    const codeLanguage = codeElement.dataset.lang;
+    const isShellLanguage = shellAliases.indexOf(codeLanguage) >= 0;
+
+    if (isShellLanguage) {
+      const [firstChar] = code;
+      code = firstChar === '$' ? code.substring(2) : code;
```

Comment:
This should be more reliable:

```suggestion
      code = code.replace(/^\$ /gm, '');
```

It will also remove the dollar sign on each line

Note that it only removes a dollar sign followed by a space. Otherwise we could potentially be looking at a variable (`$CMD arg0 arg1`...?).

Reply:
Good call out. I think I noticed a few multi-line shell examples so this solution is better. Thank you, will update!